src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_OF_FIGURES = 5;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        String color = colorSupplier.getRandomColor();
+        int figureType = random.nextInt(NUMBER_OF_FIGURES);
+        switch (figureType) {
+            case 0:
+                return new Circle(color, random.nextDouble() * 10);
+            case 1:
+                return new Square(color, random.nextDouble() * 10);
+            case 2:
+                return new Rectangle(color, random.nextDouble() * 10, random.nextDouble() * 10);

```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables for the width and length and pass them on for better code readability.